File: provider/src/main/java/com/glancebar/example/provider/ProviderApplication.java
package com.glancebar.example.provider;

import org.apache.dubbo.config.spring.context.annotation.DubboComponentScan;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ImportResource;

/**
 * DubboComponentScan开启扫描注解组件
 *
 * @author YISHEN CAI
 */
@SpringBootApplication
@ImportResource(value = { "classpath:dubbo/dubbo-provider.xml" })
@DubboComponentScan(basePackages = { "com.glancebar.example.provider.service" })
public class ProviderApplication {

    public static void main(String[] args) {
        SpringApplication.run(ProviderApplication.class);
    }
}

File: provider/src/main/kotlin/com/glancebar/example/provider/ProviderApplication.kt
package com.glancebar.example.provider

import org.apache.dubbo.config.spring.context.annotation.DubboComponentScan
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ImportResource

@SpringBootApplication
@ImportResource(
    value = [
        "classpath:dubbo/dubbo-provider.xml"
    ]
)
@DubboComponentScan(basePackages = ["com.glancebar.example.provider.service"])
class ProviderApplication

fun main(args: Array<String>) {
    runApplication<ProviderApplication>(*args)
}
